src/hotspot/os_cpu/linux_arm/atomic_linux_arm.hpp

```diff
@@ -25,27 +25,27 @@
 #ifndef OS_CPU_LINUX_ARM_ATOMIC_LINUX_ARM_HPP
 #define OS_CPU_LINUX_ARM_ATOMIC_LINUX_ARM_HPP
 
+#include "memory/allStatic.hpp"
 #include "runtime/os.hpp"
 #include "runtime/vm_version.hpp"
 
 // Implementation of class atomic
 
-struct arm_atomic_funcs {
-  int64_t (*cmpxchg_long_func)(int64_t compare_value,
-                               int64_t exchange_value,
-                               volatile int64_t *dest);
-
-  int64_t (*load_long_func)(const volatile int64_t*);
-
-  void (*store_long_func)(int64_t, volatile int64_t*);
-
-  int32_t  (*add_func)(int32_t add_value, volatile int32_t *dest);
-
-  int32_t  (*xchg_func)(int32_t exchange_value, volatile int32_t *dest);
-
-  int32_t  (*cmpxchg_func)(int32_t compare_value,
-                           int32_t exchange_value,
-                           volatile int32_t *dest);
+class ARMAtomicFuncs : AllStatic {
+public:
+  typedef int64_t (*cmpxchg_long_func_t)(int64_t, int64_t, volatile int64_t*);
```

Comment:
Is the need for typedef's due to making this a proper class? Unless these typedef's get used elsewhere they don't need to exist.